stock_quant_package_fast_move/models/stock_quant_package.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (C) 2023 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, exceptions, models
+
+
+class StockQuantPackage(models.Model):
+    _inherit = "stock.quant.package"
+
+    def action_show_package_fast_move_wizard(self):
+        """Open wizard for fast package movement."""
+        return {
+            "type": "ir.actions.act_window",
+            "name": _("Move Packages"),
+            "res_model": "stock.quant.package.fast.move.wizard",
+            "target": "new",
+            "view_id": self.env.ref(
+                "stock_quant_package_fast_move.stock_quant_package_fast_move_wizard_view_form"
+            ).id,
+            "view_mode": "form",
+            "context": self.env.context,
+        }
+
+    def _move_to_location(self, location, destination_package=None):
+        """
+        Move packages to a specified location.
+
+        Parameters:
+        - location (recordset of stock.location): The destination location.
+        - destination_package (recordset of stock.quant.package, optional):
+        Optional destination package. If provided, it must belong to the specified location.
+
+        Returns:
+        - bool: True if the move is successful.
+
+        Raises:
+        - exceptions.UserError: If the destination location is the same as the current location
+        or if the destination package does not belong to the specified location.
+        """
+        # Check if the location is different from the current location
+        if location == self[0].location_id and not destination_package:
```

Comment:
Why do you only check the location_id in the first package?

What will happen if packages from different location_ids are selected?

Reply:
Before this, validation ensures the source location is the same for all packages. - [https://github.com/OCA/stock-logistics-workflow/pull/1428/files#diff-8d3622c031c6c7da6edf3c7e1d6e83021cacb13df6f5cc2f25b518262fae59deR47](https://github.com/OCA/stock-logistics-workflow/pull/1428/files#diff-8d3622c031c6c7da6edf3c7e1d6e83021cacb13df6f5cc2f25b518262fae59deR47)

Reply:
I can't spot where is that check? And why is this necessary that they are all in the same source location?
```suggestion
        if location in self.location_id and not destination_package:
```